File: app/src/main/java/prbale/com/androidmvpdemo/login/LoginContract.java
package prbale.com.androidmvpdemo.login;

public interface LoginContract {

  // View Interface
  interface View {
    String getUsername();
    void showUsernameError(int resId);
    String getPassword();
    void showPasswordError(int resId);
    void startMainActivity();
    void showLoginError(int resId);
  }

  // Presenter / Presentation Interface
  interface Presenter {
    void onLoginClicked();
  }

  // Model - Business Logic Interface
  interface Repository {
    boolean login(String username, String password);
  }
}

File: app/src/main/java/prbale/com/androidmvpdemo/login/LoginContract.kt
package prbale.com.androidmvpdemo.login

interface LoginContract {

  // View Interface
  interface View {
    fun getUsername(): String
    fun getPassword(): String
    fun showUsernameError()
    fun showPasswordError()
    fun startMainActivity()
    fun showLoginError()
  }

  // Presenter / Presentation Interface
  interface Presenter {
    fun doLogin()
  }

  // Model - Business Logic Interface
  interface Repository {
    fun login(username: String, password: String): Boolean
  }
}
